### ISSUE TITLE
`Value Conversion Error` when attempting to create an environment using calculated (unknown) `service` blocks

### ISSUE THREAD

**Issue by patrickcping** [pingidentity/terraform-provider-pingone#651]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

Thank you for opening an issue. Please note that we try to keep the Terraform issue tracker reserved for bug reports and feature requests. For general usage questions, please see: https://www.terraform.io/community.html.

### PingOne Terraform provider Version

`v0.23.0`

### Terraform Version

N/a

### Affected Resource(s)

- `pingone_environment`

### Terraform Configuration Files
```hcl
# Copy-paste your PingOne related Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key.

# Remember to replace any account/customer sensitive information in the configuration before submitting the issue

variable "services" {
  type = list(string)
  default = [
    "DaVinci",
    "MFA",
    "SSO",
  ]
}

resource "pingone_environment" "dynamic_services" {
  name        = "Terraform Testing Dynamic Services"
  type        = "SANDBOX"
  license_id  = data.pingone_licenses.license.ids[0]

  dynamic "service" {
    for_each = toset(var.services)

    content {
      type = service.key
    }
  }
}

```

### Debug Output

```
│ Error: Value Conversion Error
│ 
│   with pingone_environment.dynamic_services,
│ An unexpected error was encountered trying to build a value. This is always an error in the provider. Please report the
│ following to the provider developer:
│ 
│ Received unknown value, however the target type cannot handle unknown values. Use the corresponding `types` package type or
│ a custom type that handles unknown values.
│ 
│ Path: 
│ Target Type: []base.environmentServiceModel
│ Suggested Type: basetypes.SetValue
```

### Expected Behavior

Provider should continue without error as HCL is correct

### Actual Behavior

Error shown

### Steps to Reproduce

1. `terraform apply`

**Comment by icsrmontano** [comment]
Hi @patrickcping. FYI, got same error when destroying the environment.

**Comment by patrickcping** [comment]
hey @icsrmontano I've released `v0.23.1` today which should fix it.  Let me know though if you still see issues

**Comment by icsrmontano** [comment]
It works now. Thank you! From: Patrick Cowland ***@***.***>Date: Saturday, November 11, 2023 at 4:42 AMTo: pingidentity/terraform-provider-pingone ***@***.***>Cc: Rafael Montano ***@***.***>, Mention ***@***.***>Subject: Re: [pingidentity/terraform-provider-pingone] `Value Conversion Error` when attempting to create an environment using calculated (unknown) `service` blocks (Issue #651)hey @icsrmontano I've released v0.23.1 today which should fix it. Let me know though if you still see issues—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>
